include/ignition/gazebo/comms/MsgManager.hh

```diff
@@ -0,0 +1,157 @@
+ *
+ */
+
+#ifndef IGNITION_GAZEBO_MSGMANAGER_HH_
+#define IGNITION_GAZEBO_MSGMANAGER_HH_
+
+#include <deque>
+#include <map>
+#include <memory>
+#include <string>
+
+#include <ignition/transport/Node.hh>
+#include <ignition/utils/ImplPtr.hh>
+#include "ignition/gazebo/config.hh"
+#include "ignition/gazebo/System.hh"
+
+namespace ignition
+{
+namespace msgs
+{
+  // Forward declarations.
+  class Dataframe;
+}
+namespace gazebo
+{
+// Inline bracket to help doxygen filtering.
+inline namespace IGNITION_GAZEBO_VERSION_NAMESPACE {
+namespace comms
+{
+
+/// \brief A queue of message pointers.
+using DataQueue = std::deque<msgs::DataframeSharedPtr>;
+
+/// \brief A map where the key is the topic subscribed to an address and
+/// the value is a publisher to reach that topic.
+using SubscriptionHandler = std::map<std::string, transport::Node::Publisher>;
+
+/// \brief All the information associated to an address.
+struct AddressContent
+{
+  /// \brief Queue of inbound messages.
+  public: DataQueue inboundMsgs;
+
+  /// \brief Queue of outbound messages.
+  public: DataQueue outboundMsgs;
+
+  /// \brief Subscribers.
+  public: SubscriptionHandler subscriptions;
+
+  /// \brief Model name associated to this address.
+  public: std::string modelName;
+};
+
+/// \brief A map where the key is an address and the value is all the
+/// information associated to each address (subscribers, queues, ...).
+using Registry = std::map<std::string, AddressContent>;
+
+/// \brief ToDo.
+class MsgManager
+{
+  /// \brief Default constructor.
+  public: MsgManager();
+
+  /// \brief Destructor.
+  public: virtual ~MsgManager();
+
+  /// \brief Add a new subscriber. It's possible to associate multiple topics
+  /// to the same address/model pair. However, the same address cannot be
+  /// attached to multiple models. When all the subscribers are removed, it's
+  /// posible to bind to this address using a different model.
+  /// \param[in] _address The subscriber address.
+  /// \param[in] _modelName The model name.
+  /// \param[in] _topic The subscriber topic.
+  /// \return True if the subscriber was successfully added or false otherwise.
+  public: bool AddSubscriber(const std::string &_address,
+                             const std::string &_modelName,
+                             const std::string &_topic);
+
+  /// \brief Add a new message to the inbound queue.
+  /// \param[in] _address The destination address.
+  /// \param[in] _msg The message.
+  public: void AddInbound(const std::string &_address,
+                          const msgs::DataframeSharedPtr &_msg);
+
+  /// \brief Add a new message to the outbound queue.
+  /// \param[in] _address The sender address.
+  /// \param[in] _msg The message.
+  public: void AddOutbound(const std::string &_address,
+                           const msgs::DataframeSharedPtr &_msg);
+
+  /// \brief Remove an existing subscriber.
+  /// \param The subscriber address.
+  /// \param The Subscriber topic.
```

Comment:
```suggestion
  /// \param[in] _address The subscriber address.
  /// \param[in] _topic The Subscriber topic.
```

Reply:
Thanks, done in [beefb6b](https://github.com/ignitionrobotics/ign-gazebo/pull/1416/commits/beefb6bf950dc48ffb783c2c17ca2357ca8ad846).